landlab/components/overland_flow/generate_overland_flow_implicit_kinwave.py

```diff
@@ -0,0 +1,320 @@
+# -*- coding: utf-8 -*-
+"""
+Landlab component for overland flow using a local implicit solution to the
+kinematic-wave approximation.
+
+Created on Fri May 27 14:26:13 2016
+
+@author: gtucker
+"""
+
+
+from landlab import Component
+from landlab.components import FlowAccumulator
+from scipy.optimize import newton
+import numpy as np
+
+
+def water_fn(x, a, b, c, d, e):
```

Comment:
Needs docstring, and perhaps more intuitive function and variable names. Looks like this is the function you use Newtons method on in order to calculate water depth.